configure

```diff
@@ -1062,6 +1067,28 @@ def configure_static(o):
         o['libraries'] += ['-static-libasan']
 
 
+def configure_gold(o):
+  try:
+    proc = subprocess.Popen(shlex.split('ld.gold') + ['-v'], stdin=subprocess.PIPE, 
```

Comment:
You don't need to use `shlex.split()` here since you're passing the static string `'ld.gold'`.

That said, it should arguably be overridable through a command line option or an environment variable, like CC and CXX are and result in `-fuse-ld=/path/to/ld.gold` being passed so that the compiler picks up the right linker.

As well, this step should probably be skipped when cross-compiling.

`configure_node()` currently contains the cross-compiling logic, that should probably be factored out somehow.  For now, making `cross_compiling` global is probably acceptable.
```diff
diff --git a/configure b/configure
index 95f103fbcb..04cbb91ea9 100755
--- a/configure
+++ b/configure
@@ -35,6 +35,9 @@ import subprocess
 import shutil
 import string
 
+# Set by configure_node().
+cross_compiling = False
+
 # gcc and g++ as defaults matches what GYP's Makefile generator does,
 # except on OS X.
 CC = os.environ.get('CC', 'cc' if sys.platform == 'darwin' else 'gcc')
@@ -837,6 +840,7 @@ def configure_node(o):
   o['variables']['target_arch'] = target_arch
   o['variables']['node_byteorder'] = sys.byteorder
 
+  global cross_compiling
   cross_compiling = (options.cross_compiling
                      if options.cross_compiling is not None
                      else target_arch != host_arch)
@@ -1402,7 +1406,7 @@ if (options.dest_os):
   flavor_params['flavor'] = options.dest_os
 flavor = GetFlavor(flavor_params)
 
-configure_node(output)
+configure_node(output)  # Sets `cross_compiling` as a side effect.
 configure_library('zlib', output)
 configure_library('http_parser', output)
 configure_library('libuv', output)
```

Reply:
thanks ! hadn't thought about cross compiling